pkg/generate/app/app.go

```diff
@@ -81,6 +85,28 @@ func nameFromGitURL(url *url.URL) (string, bool) {
 	}
 	// from path
 	if name, ok := git.NameFromRepositoryURL(url); ok {
+
+		if !kvalidation.IsDNS1123Label(name) {
+			name = strings.ToLower(name)
+
+			// remove everything except [-0-9a-z]
+			name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
```

Comment:
Name suggester does not know the context under which it is used.  It's
inappropriate for NameSuggester have to be aware of the LCD of names.
Instead, suggestions have to be resolved to a context.

On Thu, Oct 22, 2015 at 3:27 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/generate/app/app.go
> <https://github.com/openshift/origin/pull/5331#discussion_r42794466>:
>
> > @@ -81,6 +85,28 @@ func nameFromGitURL(url *url.URL) (string, bool) {
> >  	}
> >  	// from path
> >  	if name, ok := git.NameFromRepositoryURL(url); ok {
> > +
> > +		if !kvalidation.IsDNS1123Label(name) {
> > +			name = strings.ToLower(name)
> > +
> > +			// remove everything except [-0-9a-z]
> > +			name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
>
> why shouldn't a name suggester suggest valid names? suggesting a name that
> the consumer then has to further modify seems odd to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5331/files#r42794466>.
>